.gitbook/5-oracles/1-operate-oracle-nodes/4-oracle-registration.md

```diff
@@ -0,0 +1,96 @@
+The `Node Key` is used to be shared the oracle private key.
+Since the oracle private key must be shared in a very secure way, the key will be encrypted with the public key of the `Node Key` so that only the oracle can decrypt it.
+
+{% hint style="danger" %}
+The private key of `Node Key` is also stored in secure way as a sealed file, named `node_priv_key.sealed` in default.
+This `node_priv_key.sealed` file is the only clue to decrypt the encrypted oracle private key, we recommend to manage it very carefully in case that you need to restore the oracle private key again later.
+{% endhint %}
+
+If you have tried to request `register-oracle` before, previously generated `node_priv_key.sealed` would exist.
+And the app will ask, 
+
+```
+There is an existing node key.
+Are you sure to delete and re-generate node key?
+```
+
+If you're sure to re-generate the `Node Key` and re-request oracle registration, please enter `y`.
+Or, we recommend to back up the existing `node_priv_key.sealed` file.
+
+## Subscribe Approval of Registration
+
+If the transaction for request of oracle registration succeeds, oracle will start to subscribe `ApproveOracleRegistrationEvent`.
+Other oracles already registered will do some verifications of this registration by checking if:
+- correct version of oracle binary is used
+- the oracle is running inside an enclave 
+- the `Node Key` is generated inside the enclave
+- the trusted block information is valid
+
+When the registration is verified successfully, other oracles will send a transaction for approval of the oracle registration.
+The oracle catch the approval transaction, and then the oracle start to retrieve the oracle private key.
+As a result of retrieval of oracle private key, a sealed file named `oracle_priv_key.sealed` (default file name) will be generated.
+Using the sealed oracle private key, this oracle is now able to function normally as an oracle.
+
+{% hint style="danger" %}
+Like `node_priv_key.sealed`, `oracle_priv_key.sealed` is also very important for working as a valid oracle.
+Thus, we highly recommend to manage the `oracle_priv_key.sealed` file very carefully.
+In case that operator lost the `oracle_priv_key.sealed` file for any reason, it can be retrieved again using `get-oracle-key` CLI with the `node_priv_key.sealed` file, which is generated when request oracle registration above.
+{% endhint %}
+
+For any reasons, the oracle can stop before the oracle private key is retrieved. 
+In this case, you can also retrieve the oracle private key through `get-oracle-key` CLI.
+If at least one registered oracle approves the registration, this will be able to retrieve the oracle private key successfully.
+
+```bash
+docker run \
+    --device /dev/sgx_enclave \
+    --device /dev/sgx_provision \
+    -v ${ANY_DIR_ON_HOST}:/oracle \
+    ghcr.io/medibloc/panacea-oracle:latest \
+    ego run /usr/bin/oracled get-oracle-key
+```
```

Comment:
How about providing a way to check if oracle registration was successful? And, It would be good if there was a link to the `running-node` guide.

Reply:
Ok. I'll add it. Thansks!